lib/components/SelectionControl/checkbox/checkbox-group/index.tsx

```diff
@@ -0,0 +1,145 @@
+import { useState, useEffect } from "react";
+import clsx from "clsx";
+import "./checkbox-group.scss";
+import { Checkbox } from "../checkbox-single";
+import { CheckboxProps } from "../checkbox-single";
+
+interface Node extends Omit<CheckboxProps, "id" | "children"> {
+    id: string | number;
+}
+
+interface ParentNode extends Node {
+    children?: Node[];
+}
+
+interface CheckboxGroupProps {
+    className?: string;
+    checkboxGroupConfig: ParentNode[];
+}
+
+export const CheckboxGroup = ({
+    className,
+    checkboxGroupConfig,
+}: CheckboxGroupProps) => {
+    const [checkBoxItemsConfig, setCheckBoxItemConfig] =
+        useState(checkboxGroupConfig);
+
+    useEffect(() => {
+        const configCopy = structuredClone(checkBoxItemsConfig);
```

Comment:
It has 93.59% support (CanIUse). Screenshot from mdn web_docs is pinned below. Has polyfill.
In case if it's considered unreliable, I can still add lodash library to projects dependencies and use a cloneDeep() function. Still, the downside is that it requires to add the library to your project, making it an external dependency.
<img width="835" alt="Screenshot 2024-04-11 at 8 06 04 AM" src="https://github.com/deriv-com/quill-ui/assets/121025168/4219e263-63db-43f0-9d88-a3e190eae8e4">